File: algos/mini_task3.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        while l <= r:
            pivot = (l + r) // 2
            if nums[pivot] == target:
                return pivot
            elif nums[pivot] > target:
                r = pivot - 1
            elif nums[pivot] < target:
                l = pivot + 1
        return -1

# Ссылка на Leetcode: https://leetcode.com/problems/binary-search/submissions/897257324/

File: algos/mini_task3.swift
class Solution {
    func search(_ nums: [Int], _ target: Int) -> Int {
        var l: Int = 0
        var r: Int = nums.count - 1
        while l <= r {
            var pivot: Int = (l + r) / 2
            if nums[pivot] == target {
                return pivot
            }
            else if nums[pivot] > target{
                r = pivot - 1
            }
            else if nums[pivot] < target {
                l = pivot + 1
            }
        }
        return -1
    }
}